### PR TITLE
link : add support for TCP Segmentation Offload 

### PR DESCRIPTION
**Pull request by ssahani**
patch for issue #432

### CODE REVIEW

File: man/systemd.link.xml [hunk 1]
```diff
@@ -377,6 +377,14 @@
...
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>TCPSegmentationOffload=</varname></term>
+        <listitem>
+          <para>The TCP Segmentation Offload (TSO) when true enables
+          tcp segmentation offload. Takes a boolean value.
+          Defaults to "unset".</para>
+        </listitem>
+      </varlistentry
```
Comment (zonque) [inline man/systemd.link.xml]: Semaphore is right, there's a syntax error.

**Comment by ssahani** [comment]
oops fixed .

**Comment by michich** [comment]
Isn't ETHTOOL_STSO considered a legacy ethtool operation? The kernel has more generic mechanisms for querying and setting netdev features: ETHTOOL_GFEATURES, ETHTOOL_SFEATURES, ETHTOOL_GSTRINGS with ETH_SS_FEATURES.
Did you consider implementing a more generic way of configuring netdev features? TSO would be just of them.

**Comment by ssahani** [comment]
Since features we do mostly on demand I did looked at the TSO feature. But essentially you have pointed out that we can cover other features also. Could you point how we can do this. any working example ?

**Comment by michich** [comment]
For an example, see the ethtool utility itself. It uses all of the new mechanisms to implement ethtool -k and -K. But it also implements fallbacks to the old mechanisms, which you do not need to implement in systemd.

**Comment by ssahani** [comment]
Can you elaborate . I seen this part actually . I could not figure it out .

**Pull request closed without merging**